graphcanvas/graph_container.py

```diff
@@ -35,12 +35,26 @@ def do_layout(self, size=None, force=False):
             return
 
         def _apply_graphviz_layout(layout):
-            min_x = min([pos[0] for pos in layout.values()])
-            max_y = max([pos[1] for pos in layout.values()])
+            x_values = [pos[0] for pos in layout.values()]
+            y_values = [pos[1] for pos in layout.values()]
+            min_x = min(x_values)
+            max_x = max(x_values)
+            min_y = min(y_values)
+            max_y = max(y_values)
+            x_range = max_x - min_x
+            y_range = max_y - min_y
 
             for component in self.components:
-                component.x = self.width + layout[component._key][0] - min_x
-                component.y = self.height + layout[component._key][1] - max_y
+                component.x = (
+                    self.width *
+                    (layout[component._key][0] - min_x) /
+                    x_range
+                )
+                component.y = (
+                    self.height *
+                    (layout[component._key][1] - min_y) /
+                    y_range
+                )
```

Comment:
if i'm not wrong, this fixes test failures - but i don't know what these changes are doing.

Reply:
Basically, the `layout` comes back from `graphcanvas` and is x and y values scaled to whatever size `graphcanvas` wants in inches, and we want to rescale things to fit inside the container's bounds (in pixels), so this function rescales the x and y values to be the appropriately scaled values in the container's bounds space.

I'm sure the old version of this worked at some point, but we haven't been testing against it for several years (installing `pygraphviz` got very difficult for a couple years but now it's better). The old logic may have worked simply by coincidence, but it doesn't make sense to me how it would have worked looking at it now.